lib/fizzy/execute.cpp

```diff
@@ -323,15 +323,20 @@ const FuncType& function_type(const Instance& instance, FuncIdx idx)
 }
 
 bool invoke_function(const FuncType& func_type, uint32_t func_idx, Instance& instance,
-    Stack<uint64_t>& stack, int depth)
+    OperandStack& stack, int depth)
 {
     if (depth == CallStackLimit)
         return false;
 
     const auto num_args = func_type.inputs.size();
     assert(stack.size() >= num_args);
-    std::vector<uint64_t> call_args(stack.end() - static_cast<ptrdiff_t>(num_args), stack.end());
-    stack.resize(stack.size() - num_args);
+    std::vector<uint64_t> call_args;
```

Comment:
It still can be done with a constructor, for example
```suggestion
    const auto args_begin = &stack[0] - num_args + 1;
    std::vector<uint64_t> call_args(args_begin, args_begin + num_args);
```

Reply:
The `&stack[0]` is invalid for empty stack.

Reply:
Hm right, because `operator[]` dereferences it.
What do you think about adding `cbegin()`/`cend()` to `OperandStack`?

(Here it just seems so many additional calls + `if` appeared, but maybe my intuition is wrong and it's not much of a difference performance-wise. Complexity-wise it's ok I guess)

Reply:
The `&stack[0]` should be fine on it's own, UBSan should pass it. The problem is with the `assert()` inside `operator[]`.
One option is to drop the assert and move on.
I wasn't keen to add `begin()`/`end()` because the `std::vector call_args` is to be removed Soon™. But I think I will add it now.